client/blocks/plan-thank-you-card/index.jsx

```diff
@@ -0,0 +1,123 @@
+	};
+
+	componentDidMount() {
+		if ( ! this.props.plan ) {
+			this.props.fetchSitePlans( this.props.selectedSite );
+		}
+
+		if ( ! this.props.plansFetched ) {
+			this.props.requestPlans();
+		}
+	}
+
+	render() {
+		// Non standard gridicon sizes are used here because we use them as background pattern with various sizes and rotation
+		/* eslint-disable wpcalypso/jsx-gridicon-size */
+		return (
+			<div className="plan-thank-you-card">
+				<div className="plan-thank-you-card__header">
+					<Gridicon className="plan-thank-you-card__main-icon" icon="checkmark-circle" size={ 140 } />
+					<div className="plan-thank-you-card__plan-name">
+
+					</div>
+					{ ( ! this.props.planDetails ) ? (
+						<div className="plan-thank-you-card__plan-name is-placeholder"></div>
+					) : (
+						<div className="plan-thank-you-card__plan-name">
+							{ this.props.translate( '%(planName)s Plan', {
+								args: { planName: this.props.planDetails.product_name_short }
+							} ) }
+						</div>
+					) }
+					{ ( ! this.props.plan ) ? (
+						<div className="plan-thank-you-card__plan-price is-placeholder"></div>
+					) : (
+						<div className="plan-thank-you-card__plan-price">{ this.props.plan.formattedPrice }</div>
+					) }
+					<div className="plan-thank-you-card__background-icons">
+						<Gridicon icon="audio" size={ 52 } />
+						<Gridicon icon="audio" size={ 20 } />
+						<Gridicon icon="heart" size={ 52 } />
+						<Gridicon icon="heart" size={ 41 } />
+						<Gridicon icon="star" size={ 26 } />
+						<Gridicon icon="status" size={ 52 } />
+						<Gridicon icon="audio" size={ 38 } />
+						<Gridicon icon="status" size={ 28 } />
+						<Gridicon icon="status" size={ 65 } />
+						<Gridicon icon="star" size={ 57 } />
+						<Gridicon icon="star" size={ 33 } />
+						<Gridicon icon="star" size={ 45 } />
+					</div>
+				</div>
+				<div className="plan-thank-you-card__body">
+					<div className="plan-thank-you-card__heading">
+						{ this.props.translate( 'Thank you for your purchase!' ) }
+					</div>
+					<div className="plan-thank-you-card__description">
+						{ this.props.translate( 'Now that we’ve taken care of the plan, its time to see your new site.' ) }
+					</div>
+					<a
+						className="plan-thank-you-card__button"
+						href={ this.props.selectedSite.URL }>
+						{ this.props.translate( 'Visit your site' ) }
+					</a>
+				</div>
+			</div>
+		);
+		/* eslint-enable wpcalypso/jsx-gridicon-size */
+	}
+}
+
+export default connect(
+	( state, ownProps ) => {
+		let selectedSite = getSelectedSite( state );
+
+		if ( ownProps.selectedSite && ! selectedSite ) {
+			selectedSite = ownProps.selectedSite;
+		}
+
+		const plan = getCurrentPlan( state, selectedSite.ID );
+
+		let planDetails = null;
+		if ( plan ) {
+			planDetails = getPlanBySlug( state, plan.productSlug );
+		}
+
+		return {
+			selectedSite,
+			plan,
+			planDetails,
+			plansFetched: getPlans( state ).length !== 0
+		};
+	},
+	( dispatch ) => {
```

Comment:
Did you know that connect also accepts not just a function, but an [object](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)? This can help reduce boilerplate if there's nothing custom happening with our actions. We don't need `fetchSitePlans` or `requestPlans` in this PR, but in the future you could write this as:

```javascript
{
     fetchSitePlans,
     requestPlans
}
```